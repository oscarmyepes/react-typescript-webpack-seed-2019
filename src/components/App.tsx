import * as React from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { logTestSaga } from "../store/modules/test/actions";

interface IApp {
  dispatch: Dispatch;
}

export let App = ({ dispatch }: IApp) => {
  React.useEffect(() => {
    dispatch(logTestSaga("from App"));
  });
  return <div>In app</div>;
};
export default connect()(App);
