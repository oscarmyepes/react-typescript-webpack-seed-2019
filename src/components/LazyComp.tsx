import * as React from "react";

export let Lazy = () => {
  return <div>Lazy</div>;
};
export default Lazy;
