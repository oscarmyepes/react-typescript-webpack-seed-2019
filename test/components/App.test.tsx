import { shallow } from "enzyme";
import * as React from "react";
import * as renderer from "react-test-renderer";
import { App } from "../../src/components/App";

describe("test", () => {
  it("Should contain div In app", () => {
    const wrapper = shallow(<App dispatch={() => null} />);
    expect(wrapper.contains(<div>In app</div>)).toBeTruthy();
  });

  it("Should match snapshot", () => {
    const tree = renderer.create(<App dispatch={() => null} />);
    expect(tree).toMatchSnapshot();
  });
});
