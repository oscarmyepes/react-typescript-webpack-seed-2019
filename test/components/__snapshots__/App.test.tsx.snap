// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Should match snapshot 1`] = `
<div>
  In app
</div>
`;
